QUARKUS-451.md

```diff
@@ -0,0 +1,42 @@
+# QUARKUS-451 - Testing and verifying OpenShift Kafka with Quarkus
+
+The goal of this test plan is to cover the main points of conflicts between Quarkus and Kafka. 
+
+JIRA link: https://issues.redhat.com/browse/QUARKUS-451
+
+## Scope of the testing
+Following actions were taken to ensure test coverage and automation for reactive kafka applications:
+ - Kafka producer with JSON / AVRO body messages.
+ - Kafka consumer with JSON / AVRO body messages.
+ - Kafka Streams with JSON body messages and windowed.
+ - Was tested over Kafka Strimzi and RH AMQ Streams.
+ - Platform Openshift, Java 11.
+
+### Impact on testsuites and testing automation:
+ - OpenShift TS framework needs to be enhanced to support OpenShift kafka
+   - Would be great if in the future we support some kind of openshift/k8s templating. So install some specifict Kafka (provider/version) would be as simple as apply a template. 
+
+### Impact on resources:
+ - More services will be running on OpenShift cluster, impact on CPU/memory should be minimal (if you don't setup Kafka/zookeper/registry with a persistent volume).
+
+## Getting familiar with the feature
+Following actions were taken to ensure familiarity:
+ - Focus on exploratory testing of the features
+ - Ensure good user experience and simplicity of use
+
+## Automated test development
+
+Two scenarios has been setup:
+- [KStreams Scenario](https://github.com/quarkus-qe/quarkus-openshift-test-suite/tree/master/messaging/kafka-streams-reactive-messaging): There is an EventsProducer that generate login status events every 100ms. A Kafka stream called WindowedLoginDeniedStream will aggregate these events in fixed time windows of 3 seconds. So if the number of wrong access excess a threshold, then a new alert event is thrown. All aggregated events(not only unauthorized) are persisted.
+
+- [Kafka AVRO Scenario](https://github.com/quarkus-qe/quarkus-openshift-test-suite/pull/140): There is an EventsProducer that generates stock price events every 1s. The events are typed by an AVRO schema.  
+A Kafka consumer will read these events serialized by AVRO and change a `status` property to `COMPLETED`. 
+The streams of completed events will be exposed through an SSE endpoint.
+
+## Advanced topics for test development
+The following advanced topics are listed for future consideration. There should be dedicated feature requirements and/or enhancements for these specific areas.
+ - Security testing
+ - Recovery/Failover testing
+ - Performance/Scalability testing
+ - Backward compatibility testing
```

Comment:
What did you have in mind? Backward compatibility of clients or brokers? Should that be concern of Quarkus testing?